pkg/builder_test.go

```diff
@@ -0,0 +1,2277 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package statediff_test
+
+import (
+	"bytes"
+	"math/big"
+	"sort"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/rlp"
+
+	statediff "github.com/vulcanize/eth-statediff-service/pkg"
+	"github.com/vulcanize/eth-statediff-service/pkg/testhelpers"
+)
+
+// TODO: add test that filters on address
+var (
+	contractLeafKey                                        []byte
+	emptyDiffs                                             = make([]statediff.StateNode, 0)
+	emptyStorage                                           = make([]statediff.StorageNode, 0)
+	block0, block1, block2, block3, block4, block5, block6 *types.Block
+	builder                                                statediff.Builder
+	miningReward                                           = int64(2000000000000000000)
+	minerAddress                                           = common.HexToAddress("0x0")
+	minerLeafKey                                           = testhelpers.AddressToLeafKey(minerAddress)
+	workerCounts                                           = []uint{0, 1, 2, 4, 8}
```

Comment:
Could you add to this list values `3` and `5` and check tests? I would like to check how it will work with an odd amount of workers

Reply:
The iterator is actually restricted (with a panic) to splitting by powers of 2 for now, just for simplicity. Can be changed later.